### PR TITLE
Setup config class based on toml

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Please see https://issuetracker.deltares.nl/browse/IMOD6-348

### CODE REVIEW

File: imod_coupler/config.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        It also checks for environment variables used to override the config path.
+
+        Raises:
+            ConfigError: If required is specified and the object is not found
+                (and there is no default value provided), this error will be raised
+        """
+
+        # Create the environment variable string
+        # e.g. IMODC__KERNELS__MODFLOW6__DLL
+        env_variable = "IMODC"
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply imod_coupler/config.py:94]: Good idea.
Unfortunately, this doesn't work since path is not a list of str, but a list of str and int.

**Review by rleander — APPROVED** [review]
Review OK

File: imod_coupler/config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rleander) [reply imod_coupler/config.py:94]: Why is there int in path .... ?

[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply imod_coupler/config.py:94]: Because, toml supports arrays and arrays are indexed with int's not strings

**Pull request merged**